src/qe/mbp/mbp_term_graph.h

```diff
@@ -151,6 +195,44 @@ namespace mbp {
         void  add_model_based_terms(model& mdl, expr_ref_vector const& terms);
         expr* rep_of(expr* e);
 
+      // new methods by IG
+        using deqs = bit_vector;
+        struct add_deq_proc {
+          uint64_t m_deq_cnt =
```

Comment:
This comment too, maybe add an assertion that m_deq_cnt is in range when adding a new disequality